presto-native-execution/presto_cpp/main/common/Configs.h

```diff
@@ -327,4 +329,41 @@ class NodeConfig : public ConfigBase {
       const std::function<uint64_t()>& defaultNodeMemoryGb = nullptr) const;
 };
 
+/// Used only in the single instance as the source of the initial properties for
+/// velox::QueryConfig. Not designed for actual property access during a query
+/// run.
+/// Values can be modified via Server Operation command if the SystemConfig has
+/// kMutableConfig option set to true (so this config must be created after the
+/// SystemConfig has been initialized).
+class BaseVeloxQueryConfig {
+ public:
+  BaseVeloxQueryConfig();
+
+  static BaseVeloxQueryConfig* instance();
+
+  /// If this config is mutable.
+  bool isMutable() const {
+    return mutable_;
+  }
+
+  /// Returns copy of the config values map.
+  std::unordered_map<std::string, std::string> values() const {
```

Comment:
I'm still not quite clear on how we would like this class to work together with VeloxConfig to mutate VeloxConfig on the fly. Maybe wait for the next PR. Should here be std::unordered_map<>& reference?

Separately, class name BaseVeloxQueryConfig is not giving me much info. Do you think it's better to call it delegate or prototype, depending on how we're going to use it?